renderer/src/common/hooks/use-form-tab-state.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react'
+
+interface FieldTabMapping {
+  field: string
+  tab: string
+}
+
+interface UseFormTabStateOptions {
+  fieldTabMap: FieldTabMapping[]
+  defaultTab?: string
+}
+
+interface UseFormTabStateReturn {
+  activeTab: string
+  setActiveTab: (tab: string) => void
+  showFieldError: (fieldName: string) => void
+  activateTabWithError: (errors: Record<string, unknown>) => void
+}
+
+export function useFormTabState({
```

Comment:
note, this just extracts existing common functionality, but it's already existing code. 

in a later design the tabs will likely be removed and replaced with a multi-step form where likely there will be a validation at each step.

however, this does **not** mean that this hook will no longer be needed. this is because some errors, such as duplicated MCP name will only be discovered at the final submission. So this same logic will likely be kept long term, perhaps with small adjustments

Reply:
also perhaps there is a more sophisticated/idiomatic way to do it with react-hook-forms :thinking: I didn't want to do a mega refactor here, but I am open to suggestions on that and I can implement it in a follow-up PR :thinking:

Reply:
I think the logic makes sense, I would not expect that react-hook-forms can be aware that there are form in different tabs, it is up to us to handle it, and this hook makes sense. 
Maybe, I would add a simple unit test